src/bootstrap/src/ferrocene/code_coverage.rs

```diff
@@ -96,7 +124,23 @@ pub(crate) fn generate_coverage_report(builder: &Builder<'_>) {
         FerroceneCoverageFor::Library => {
             let mut instrumented_binaries = vec![];
             let out_dir = builder.cargo_out(state.compiler, Mode::Std, state.target).join("deps");
-            for res in std::fs::read_dir(out_dir).expect("cannot read deps directory") {
+
+            let res_doctest_bins = std::fs::read_dir(&paths.doctests_bins_dir);
+
+            if builder.doc_tests != DocTests::No && res_doctest_bins.is_err() {
+                panic!("cannot read doctests bins directory")
+            }
+
+            let doctests_bins = res_doctest_bins.ok().into_iter().flat_map(|read_dir| {
```

Comment:
you append these bins even if `Doctests::No` is set. i worry this will run into caching bugs. please explicitly do an `if Doctests { read_dir() } else { vec![] }` rather than the after-the-fact error checking.